man/systemd.exec.xml

```diff
@@ -248,6 +248,42 @@
         <xi:include href="system-only.xml" xpointer="singular"/></listitem>
       </varlistentry>
 
+      <varlistentry>
+        <term><varname>MountPaths=</varname></term>
+
+        <listitem><para>This setting is similar to <varname>RootImage=</varname> in that it mounts a file
+        system hierarchy from a block device node or loopback file, but the destination directory can be
+        specified as well as mount options. This option expects a whitespace separated list of mount
+        definitions. Each definition consists of a colon-separated triple of source path, destination directory
+        and optional option string. Each mount definition may be prefixed with <literal>-</literal>, in which
+        case it will be ignored when its source path does not exist. The source argument is a path to a block
+        device node or regular file. The device node or file system image file needs to contain a file system
+        without a partition table. Refer to
```

Comment:
why without a part table? I mean, sooner or later one might want verity/… so two partitions start to make sense, and then it allowing that makes sense?

I would just use the usual dissect logic here, and specifiy DISSECT_IMAGE_REQUIRE_ROOT so that a GPT or MBR image with a single generic fs partition is accepted too.

Reply:
@poettering sorry I think there's some confusion due to the way I structured the PR - the first commits is, unchanged, from https://github.com/systemd/systemd/pull/14451 - the reason for that, as stated in the description, is to give full credit to Topi in the git log, since he did the first implementation. Then I added my own commit, with all the changes requested in the initial PR that is stalled - including moving from bare mount() to using the dissect login, which as you requested can handle all of that.

But it seems you prefer to review commit-by-commit, which is fair - so I have now squashed, and added a clear message and link to the original PR and mention of the original author in the commit message as an alternative.